hermes/probe/probe.go

```diff
@@ -149,22 +140,24 @@ func (mp *Probe) runProbe(ctx context.Context, metricChan chan<- *cpmetrics.Even
 	var wg sync.WaitGroup
 	for _, t := range mp.targets {
 		wg.Add(1)
-		go func(t *Target) {
+		go func(t *target.Target) {
 			defer wg.Done()
-			var err error
-			if t.LatencyMetrics, err = metrics.NewMetrics(mp.config, t.Target); err != nil {
-				mp.logger.Errorf(err.Error())
-				return
+			if t.LatencyMetrics == nil {
+				var err error
+				if t.LatencyMetrics, err = metrics.NewMetrics(mp.config, t.Target); err != nil {
+					mp.logger.Errorf(err.Error())
```

Comment:
This should be either
`mp.logger.Errorf(err.Error())`
or preferably
`mp.logger.Errorf("NewMetrics(...) failed: %v", err)`

In general: use `Error()` if there's only a single argument (there's no formatting).

Reply:
Okay, I've changed it to use `mp.logger.Errorf("NewMetrics...", err)`.